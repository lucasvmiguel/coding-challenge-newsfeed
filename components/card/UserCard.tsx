import Link from 'next/link'
import styled from 'styled-components'

import { Fellowship } from '../../graphql/client/__generated__/globalTypes'

import Card, { Icon, Columns, ColumnLeft, ColumnRight } from './Card'
import Markdown from '../Markdown'

type Project = {
  id: number;
  name: string;
  icon_url: string;
}

export type User = {
  id: number;
  name: string;
  bio: string;
  fellowship: Fellowship;
  avatar_url: string;
  projects: Project[];
}

type Props = {
  user: User;
}

export default function UserCard({ user }: Props) {
  return (
    <Card>
      <Columns>
        <ColumnLeft>
          <Avatar src={user.avatar_url} />
        </ColumnLeft>
        <ColumnRight>
          <h2>{user.name}</h2>
          <p>Fellowship: {user.fellowship}</p>
          <Markdown>{user.bio}</Markdown>
          {!!user.projects.length && (
            <>
              <h3>Projects:</h3>
              {user.projects.map(p => (
                <ProjectItem key={p.id} project={p} />
              ))}
            </>
          )}
        </ColumnRight>
      </Columns>
    </Card>
  )
}

const Avatar = styled.img`
  background-color: rgba(0, 0, 0, 0.1);
  border-radius: 10px;
`

function ProjectItem({ project }: { project: Project }) {
  return (
    <ProjectContainer>
      <ProjectColumnLeft>
        <Icon src={project.icon_url} />
      </ProjectColumnLeft>
      <ProjectColumnRight>
        <Link href={`/projects/${project.id}`}>
          {project.name}
        </Link>
      </ProjectColumnRight>
    </ProjectContainer>
  )
}

const ProjectContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin-bottom: 1rem;
`

const ProjectColumnLeft = styled.div`
  flex-basis: 2rem;
  flex-shrink: 0;
  flex-grow: 0;
  margin-right: 1rem;
`

const ProjectColumnRight = styled.div`
  flex-basis: 3rem;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
`
