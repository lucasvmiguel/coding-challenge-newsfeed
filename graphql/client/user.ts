import { gql } from '@apollo/client'

export const USER_QUERY = gql`
  query user($id: Int!) {
    user(id: $id) {
      id
      name
      bio
      fellowship
      avatar_url
      projects {
        id
        name
        icon_url
      }
    }
  }
`