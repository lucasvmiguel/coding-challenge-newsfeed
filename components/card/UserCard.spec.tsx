import renderer from 'react-test-renderer'

import { user } from '../../test/mocks/user'

import UserCard from './UserCard'

it('renders UserCard component correctly - snapshot', () => {
  const tree = renderer
    .create(<UserCard user={user} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
