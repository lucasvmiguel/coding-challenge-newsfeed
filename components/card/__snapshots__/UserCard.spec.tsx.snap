// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserCard component correctly - snapshot 1`] = `
<div
  className="sc-bdfBQB eDCjUP"
>
  <div
    className="sc-dlfnuX iOZAXF"
  >
    <div
      className="sc-hKgJUU jkblDv"
    >
      <img
        className="sc-gKseQn hvAGSQ"
        src="icon.svg"
      />
    </div>
    <div
      className="sc-eCstlR evEecB"
    >
      <h2>
        name testing
      </h2>
      <p>
        Fellowship: 
        founders
      </p>
      <div
        className="sc-jSgvzq"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>desc testing</p>
",
          }
        }
      />
      <h3>
        Projects:
      </h3>
      <div
        className="sc-iBPTik ddGxJq"
      >
        <div
          className="sc-fubCzh kGMoIv"
        >
          <img
            className="sc-gsTEea iRQMQw"
            src="icon2.svg"
          />
        </div>
        <div
          className="sc-pGacB dXoMBx"
        >
          <a
            href="/projects/2"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            user testing
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
